python/mxnet/ndarray/ndarray.py

```diff
@@ -2508,6 +2619,88 @@ def copy(self):
         """
         return self.copyto(self.context)
 
+    def slice_assign_scalar(self, value, begin, end, step):
```

Comment:
Why a wrapper on top of `_internal._slice_assign_scalar` is needed?

Reply:
To enable polymorphism and reuse code as much as possible. I reused NDArray's `_set_nd_basic_indexing` function in ndarray class and wrapping `slice_assign_scalar` and `slice_assign` as class methods can let python automatically call NDArray operators in NDArray processing, and ndarray ones in ndarray processing. If you feel such handling confusing, I can also copy-paste the entire `_set_nd_basic_indexing` function to multiarray.py.

Reply:
Same for some other operators like `_full` and `scatter_set_nd`.